### PR TITLE
Breaking: Update APIs to support axe-core

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

Add and update APIs to better align with DOM standards.
Updates aim to provide sufficient support to run axe-core.

### CODE REVIEW

File: packages/utils-dom/src/create-element.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { ElementData } from './types';
+
+const getElementType = (name: string): typeof HTMLElement => {
+    switch (name) {
+        case 'a':
+            return HTMLAnchorElement;
+        case 'body':
+            return HTMLBodyElement;
+        case 'html':
+            return HTMLHtmlElement;
```
Comment (molant) [inline packages/utils-dom/src/create-element.ts]: 🤯

File: packages/utils-dom/src/node.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                return this._node.name.toUpperCase();
+            case 'text':
+                return '#text';
+            /* istanbul ignore next */
+            default:
+                throw new Error(`Unrecognized node type ${(this._node as any).type}`);
+        }
+    }
+
+    public get nodeType(): number {
```
Comment (molant) [inline packages/utils-dom/src/node.ts]: Will be good to point to the spec just for reference

Comment (antross) [reply packages/utils-dom/src/node.ts]: I'd prefer to reference MDN (which itself links to the related spec or specs). I find it to be a better place to start to understand the basics of what an API does. What do you think?

E.g. for this case the link would be
https://developer.mozilla.org/docs/Web/API/Node/nodeType

Comment (molant) [reply packages/utils-dom/src/node.ts]: SGTM!

**Pull request merged**